### ISSUE TITLE
Make testing of resource disappearance easier to test

### ISSUE THREAD

**Issue by radeksimko** [hashicorp/terraform-plugin-sdk#9]
Testing whether Read or Exists implementations are correctly removing resource from state, when it is gone currently requires significant number of LOC and leads to lack of such tests, which in turn then leads to bugs like this https://github.com/hashicorp/terraform/issues/18931

The goal is to explore options for making such tests easier to write and generally making it easier to test such scenarios.

----

From @bflad :

This would save quite a few bug reports!

Would it make sense if this was a new boolean on `TestStep`? e.g. `ExpectsRecreationIfDeleted` (fixing my poor naming choice). It would automatically call the resource `Delete` function then ideally something similar to `ExpectsNonEmptyPlan`, but  expecting a new resource in the difference rather than just any plan difference.

There are actually reasons to go beyond just a boolean here and instead accept a resource name from the Terraform state. Here's a great example: https://github.com/terraform-providers/terraform-provider-aws/issues/5914

Basically, ELBs can have children load balancer policies, which are wholly dependent on the ELB parent existing. In this case, for this acceptance testing I will actually need two separate TestStep's to verify the resource correctly catches both of the errors returned by the API:
* One for removal of the load balancer policy (PolicyNotFound)
* One for removal of the ELB (LoadBalancerNotFound)

**Comment by kmoe** [comment]
Duplicate of #236.
